### DISCUSSION TITLE
Reading meshes from StringIO or BytesIO objects

### DISCUSSION THREAD

**Discussion by gatling-nrl** [nschloe/meshio#1348] [Q&A]
I've got the text data of a gmsh mesh in memory and want to feed it to meshio using either `StringIO` or `BytesIO` instead of writing it out to a file and passing the file name. This fails and I've traced the problem down to this line:

https://github.com/nschloe/meshio/blob/0138cc8692b806b44b32d344f7961e8370121ff7/src/meshio/gmsh/_gmsh41.py#L122

And this [open (and ancient) issue in numpy](https://github.com/numpy/numpy/issues/2230).

This sits somewhere in the grey space between bug and feature request. Is there any workaround in `meshio`? There is a workaround suggested in the numpy issue comments, but requires changing meshio.

**Comment by keurfonluu** [comment]
Hi @gatling-nrl ,

I have an opened PR for this issue: https://github.com/nschloe/meshio/pull/1305
Does it work for you?
